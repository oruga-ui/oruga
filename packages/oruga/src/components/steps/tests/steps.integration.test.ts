import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";
import { shallowRef } from "vue";

import OSteps from "@/components/steps/Steps.vue";
import OStepItem from "@/components/steps/StepItem.vue";
import OButton from "@/components/button/Button.vue";

describe("OSteps with OStepItem tests", () => {
    enableAutoUnmount(afterEach);

    function genTestcomponent(stepProps, itemProps) {
        return {
            components: {
                OSteps,
                OStepItem,
            },
            data: () => ({ stepProps, itemProps }),
            template: `<OSteps v-bind="stepProps">
                    <OStepItem v-bind="itemProps" />
                  </OSteps>
                `,
        };
    }

    test("test render correctly", () => {
        const wrapper = mount(
            genTestcomponent(
                {},
                {
                    value: "1",
                    step: "1",
                    label: "Account",
                    content: "ABC",
                },
            ),
        );
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();

        expect(wrapper.attributes("data-oruga")).toBe("steps");

        const step = wrapper.find('[data-oruga="steps-item"]');
        expect(step.exists()).toBeTruthy();
        expect(step.text()).toBe("ABC");
    });

    test("render item with component prop correctly", () => {
        const wrapper = mount(
            genTestcomponent(
                {},
                {
                    value: "1",
                    step: "1",
                    component: shallowRef(OButton),
                    props: { label: "MyButton" },
                },
            ),
        );

        const step = wrapper.find('[data-oruga="steps-item"]');
        expect(step.exists()).toBeTruthy();
        const button = step.find('[data-oruga="button"]');
        expect(button.text()).toBe("MyButton");
    });
});
