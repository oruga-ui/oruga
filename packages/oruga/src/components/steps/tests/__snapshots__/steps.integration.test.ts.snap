// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OSteps with OStepItem tests > test render correctly 1`] = `
"<div class="o-steps" data-oruga="steps">
  <ol class="o-steps__nav o-steps__nav--animated" role="tablist" aria-orientation="horizontal"></ol>
  <section class="o-steps__content">
    <!--
                @slot Place step items here
            -->
    <transition-stub appear="false" persisted="false" css="true">
      <div id="tabpanel-1" class="o-steps__panel" data-id="steps-1" data-oruga="steps-item" aria-labelledby="tab-1" aria-roledescription="item">
        <!-- 
                    @slot Step item content
                -->
        <!-- injected component -->
        <!-- default content prop -->ABC
      </div>
    </transition-stub>
  </section>
  <!--
            @slot Override step navigation
            @binding {{disabled: boolean, action: (): void }} previous - previous button configs
            @binding {{disabled: boolean, action: (): void }} next - next button configs
        -->
  <nav class="o-steps__navigation"><button disabled="" type="button" class="o-btn o-btn--disabled" role="button" data-oruga="button" aria-label="Previous"><span class="o-btn__wrapper"><span class="o-icon o-btn__icon o-btn__icon-left" data-oruga="icon"><!-- custom icon component --><!-- native css icon --><i class="mdi mdi-chevron-left mdi-24px"></i></span>
      <!--v-if-->
      <!--v-if--></span>
    </button><button disabled="" type="button" class="o-btn o-btn--disabled" role="button" data-oruga="button" aria-label="Next"><span class="o-btn__wrapper"><span class="o-icon o-btn__icon o-btn__icon-left" data-oruga="icon"><!-- custom icon component --><!-- native css icon --><i class="mdi mdi-chevron-right mdi-24px"></i></span>
      <!--v-if-->
      <!--v-if--></span>
    </button></nav>
</div>"
`;
